src/odemis/driver/ueye.py

```diff
@@ -1078,6 +1134,36 @@ def _buffer_as_array(self, mem, md):
 
         return model.DataArray(na, md)
 
+    def _set_aoi(self, sensor_res: tuple, max_res: tuple):
+        """
+        :param sensor_res: tuple(int width, int height) The maximum resolution of the camera sensor
+        :param max_res: tuple(int width, int height) The maximum requested resolution of the image
+
+        note: This function is not supported by the camera models USB 3 uEye XC and XS
+        """
+        # sets the position and size of the image by using an object of the IS_RECT type.
+        # make sure the AOI rect region is centered compared to the sensor resolution
+        if self._transpose:
+            if abs(self._transpose[0]) > abs(self._transpose[1]):
+                self.rect_aoi.s32X = (sensor_res[1] - max_res[1]) // 2
+                self.rect_aoi.s32Y = (sensor_res[0] - max_res[0]) // 2
+                self.rect_aoi.s32Width = max_res[1]
+                self.rect_aoi.s32Height = max_res[0]
+            else:
+                self.rect_aoi.s32X = (sensor_res[0] - max_res[0]) // 2
+                self.rect_aoi.s32Y = (sensor_res[1] - max_res[1]) // 2
+                self.rect_aoi.s32Width = max_res[0]
+                self.rect_aoi.s32Height = max_res[1]
+        else:
+            self.rect_aoi.s32X = (sensor_res[0] - max_res[0]) // 2
+            self.rect_aoi.s32Y = (sensor_res[1] - max_res[1]) // 2
+            self.rect_aoi.s32Width = max_res[0]
+            self.rect_aoi.s32Height = max_res[1]
+
+        # TODO use the return value to see if the command was successful??
```

Comment:
No need. The UEyeDLL already takes care of checking the return value and convert it to a Exception (UEyeError).